src/app/api/issues/route.js

```diff
@@ -4,30 +4,44 @@ import fs from 'fs';
 import yaml from 'js-yaml';
 
 export async function GET() {
-  try {
-    const issues = await prisma.Issue.findMany({
-      orderBy: {
-        id: 'desc',
-      },
-      include: {
-        user: {
-          select: {
-            login: true,
+  const user = await getCurrentUser();
+  if (!user) {
+    console.log(`#########\n UNAUTORISED \n#########`);
+    return new Response({}, { status: 401 });
+  } else {
+    try {
+      let issues = await prisma.Issue.findMany({
+        include: {
+          user: {
+            select: {
+              login: true,
+            },
           },
         },
-      },
-    });
-    return new Response(JSON.stringify(issues), {
-      status: 200,
-    });
-  } catch (error) {
-    console.log(`#########\n ${error.message} \n#########`);
-    return new Response({}, { status: 500 });
+      });
+      issues = issues.sort((a, b) => {
+        if (a.status === b.status) return 0;
+        if (a.status === 'resolved') return -1;
+        if (b.status === 'resolved') return 1;
+        if (a.status === 'ongoing') return -1;
+        if (b.status === 'ongoing') return 1;
+        else return 0;
+      });
+      if (user.admin === false) {
+        issues = issues.filter((issue) => issue.user.login === user.login);
```

Comment:
I think it would be better to use Prisma request to filter out, what do you think?

Reply:
good point, back then I thought it'd be extra security but the more I repeat it in my mind the less sense it makes